cmd/abapEnvironmentRunAUnitTest.go

```diff
@@ -396,14 +398,51 @@ func parseAUnitResult(body []byte, aunitResultFileName string) (err error) {
 				log.Entry().Debugf("The following test has been skipped: %s: %s", skipped.Message, skipped.Text)
 			}
 		}
+		if generateHTML == true {
+			htmlString := generateHTMLDocumentAUnit(parsedXML)
+			htmlStringByte := []byte(htmlString)
+			aUnitResultHTMLFileName := strings.Trim(aunitResultFileName, ".xml") + ".html"
+			err = ioutil.WriteFile(aUnitResultHTMLFileName, htmlStringByte, 0644)
+			if err != nil {
+				return fmt.Errorf("Writing HTML document failed: %w", err)
+			}
+			log.Entry().Info("Writing " + aUnitResultHTMLFileName + " file was successful")
+			reports = append(reports, piperutils.Path{Target: aUnitResultHTMLFileName, Name: "ATC Results HTML file", Mandatory: true})
+		}
 	}
 	//Persist findings afterwards
-	var reports []piperutils.Path
 	reports = append(reports, piperutils.Path{Target: aunitResultFileName, Name: "AUnit Results", Mandatory: true})
 	piperutils.PersistReportsAndLinks("abapEnvironmentRunAUnitTest", "", reports, nil)
 	return nil
 }
 
+func generateHTMLDocumentAUnit(parsedXML *AUnitResult) (htmlDocumentString string) {
+	htmlDocumentString = `<!DOCTYPE html><html lang="en" xmlns="http://www.w3.org/1999/xhtml"><head><title>AUnit Results</title><meta http-equiv="Content-Type" content="text/html; charset=UTF-8" /><style>table,th,td {border-collapse:collapse;}th,td{padding: 5px;text-align:left;font-size:medium;}</style></head><body><h1 style="text-align:left;font-size:large">AUnit test run results</h1><table style="border: 0px solid black"><tr><th>Run title</th><td style="padding-right: 20px">` + parsedXML.Title + `</td><th>System</th><td style="padding-right: 20px">` + parsedXML.System + `</td><th>Client</th><td style="padding-right: 20px">` + parsedXML.Client + `</td><th>ExecutedBy</th><td style="padding-right: 20px">` + parsedXML.ExecutedBy + `</td><th>Duration</th><td style="padding-right: 20px">` + parsedXML.Time + `s</td><th>Timestamp</th><td style="padding-right: 20px">` + parsedXML.Timestamp + `</td></tr><tr><th>Failures</th><td style="padding-right: 20px">` + parsedXML.Failures + `</td><th>Errors</th><td style="padding-right: 20px">` + parsedXML.Errors + `</td><th>Skipped</th><td style="padding-right: 20px">` + parsedXML.Skipped + `</td><th>Asserts</th><td style="padding-right: 20px">` + parsedXML.Asserts + `</td><th>Tests</th><td style="padding-right: 20px">` + parsedXML.Tests + `</td></tr></table><br><br><table style="width:100%"><tr><th>Severity</th><th>File</th><th>Message</th><th>Type</th><th>Text</th></tr>`
```

Comment:
Would it be possible to format the line so that it can be read more easily?

Reply:
New lines would add the following characters to the string: `+\n` which would result in a wrong format. Adding new strings to the `htmlDocumentString` variable in new lines would not make it more readable IMO.